extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/quantiles/DoublesSketchToHistogramPostAggregator.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.datasketches.quantiles;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.AggregatorUtil;
+import io.druid.query.aggregation.PostAggregator;
+import io.druid.query.cache.CacheKeyBuilder;
+
+import com.yahoo.sketches.quantiles.DoublesSketch;
+
+public class DoublesSketchToHistogramPostAggregator implements PostAggregator
+{
+
+  private final String name;
+  private final PostAggregator field;
+  private final double[] splitPoints;
+
+  @JsonCreator
+  public DoublesSketchToHistogramPostAggregator(
+      @JsonProperty("name") final String name,
+      @JsonProperty("field") final PostAggregator field,
+      @JsonProperty("splitPoints") final double[] splitPoints)
+  {
+    this.name = Preconditions.checkNotNull(name, "name is null");
+    this.field = Preconditions.checkNotNull(field, "field is null");
+    this.splitPoints = Preconditions.checkNotNull(splitPoints, "array of split points is null");
+  }
+
+  @Override
+  public Object compute(final Map<String, Object> combinedAggregators)
+  {
+    final DoublesSketch sketch = (DoublesSketch) field.compute(combinedAggregators);
+    final double[] histogram = sketch.getPMF(splitPoints);
+    for (int i = 0; i < histogram.length; i++) {
+      histogram[i] *= sketch.getN();
+    }
+    return histogram;
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public PostAggregator getField()
+  {
+    return field;
+  }
+
+  @JsonProperty
+  public double[] getSplitPoints()
+  {
+    return splitPoints;
+  }
+
+  // comparing histograms doesn't make much sense, so this comparator pretends that everything is equal
```

Comment:
I think I was told that throwing here would be a bad idea. And comparing by total count doesn't make much sense. I think we can just do nothing.

Reply:
Hmm, would you tell me why it's a bad idea? I think it's better because ```getComparator()``` is used for limitting the number of results like in TopN query, and users can get completely different results even though they run the same query multiple times.
Well, a better way is to check ```LimitSpec``` is specified with ```DoublesSketchToHistogramPostAggregator``` together before query execution, but I think it's beyond the scope of this PR.

Reply:
I heard that there are some systems like dashboard frontends and such, which generate queries automatically, and always use some ordering. We don't want to break compatibility with such systems. And providing some ordering, which doesn't make much sense, also sounds like a bad idea.

Reply:
I agree on that providing some ordering doesn't make sense. 
But, I think there is no compatibility issue because this is a new feature and the systems generating Druid queries can change their logic to not include ordering for this new postAggregator.

Reply:
Yes, this particular aggregator is a new feature, but we don't want it to have some properties, which would be obstacles to integration with other systems like Hive or Pivot. These systems just assume some ordering by default. I don't think we are in a position to dictate the rules.

Reply:
@jihoonson not comparable does not means its an error, IMO it means we can expect random ordered set instead of sorted ordered set. 
Some  clients (including pivot) add a default limit spec with order by on the selected metric, not sure if we want to fail the request or return a random ordered set ?

Reply:
What I'm concerned with is, when people get a result of a query with ordering and a limit, they expect an ordered top-n result. But we cannot guarantee that the result is ordered in some order with this post aggregator, so I think this is an error of unsupported feature.

I understand what you guys are concerned with, but I think we can help the clients to do the right thing by like explicitly specifying that ordering by this post aggregator is not supported in the release note.

Reply:
> Yes, this particular aggregator is a new feature, but we don't want it to have some properties, which would be obstacles to integration with other systems like Hive or Pivot.  

I don't think this is an obstacle to other ecosystems of Druid. This post aggregator is newly added in this patch, and they can add some special logic to handle this post aggregator when they decide to support it.

> These systems just assume some ordering by default. I don't think we are in a position to dictate the rules.

I'm not sure why you think so. Every system evolves as time goes by, and their ecosystems should be evolved together. Assuming every result can be ordered may be true so far, but it becomes wrong with this post aggregator. There is a recent example similar to this case. We recently added support for numeric dimensions. Druid's ecosystems can assume that every dimension has the string type before, but now it's a wrong assumption.